retrieval/target.go

```diff
@@ -155,34 +151,61 @@ type target struct {
 	// Channel to buffer ingested samples.
 	ingestedSamples chan clientmodel.Samples
 
-	url string
 	// What is the deadline for the HTTP or HTTPS against this endpoint.
 	deadline time.Duration
-	// Any base labels that are added to this target and its metrics.
-	baseLabels clientmodel.LabelSet
 	// The HTTP client used to scrape the target's endpoint.
 	httpClient *http.Client
 
-	// Mutex protects lastError, lastScrape, state, and baseLabels.
+	// Mutex protects the members below.
 	sync.RWMutex
+
+	url *url.URL
+	// Any base labels that are added to this target and its metrics.
+	baseLabels clientmodel.LabelSet
+	// The current health state of the target.
+	state TargetState
+	// The last encountered scrape error, if any.
+	lastError error
+	// The last time a scrape was attempted.
+	lastScrape time.Time
 }
 
 // NewTarget creates a reasonably configured target for querying.
-func NewTarget(url string, deadline time.Duration, baseLabels clientmodel.LabelSet) Target {
+func NewTarget(host string, cfg config.JobConfig, baseLabels clientmodel.LabelSet) Target {
```

Comment:
I wonder if instead of `host` we should in general call this `address`, as the standard lib does the same for host-port pairs: http://golang.org/pkg/net/#Dial